src/machine/machinectl.c

```diff
@@ -117,9 +118,56 @@ static int compare_machine_info(const void *a, const void *b) {
         return strcmp(x->name, y->name);
 }
 
+static int get_os_name(sd_bus *bus, const char *name, char **out)
+{
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        const char *k, *v;
+        int r;
+
+        assert(bus);
+        assert(name);
+        assert(out);
+
+        r = sd_bus_call_method(bus,
+                "org.freedesktop.machine1",
+                "/org/freedesktop/machine1",
+                "org.freedesktop.machine1.Manager",
+                "GetMachineOSRelease",
+                NULL, &reply, "s", name);
+
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_enter_container(reply, 'a', "{ss}");
+
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        *out = NULL;
+        while ((r = sd_bus_message_read(reply, "{ss}", &k, &v)) > 0) {
+                if (streq(k, "PRETTY_NAME")) {
+                        if (_unlikely_(*out != NULL))
+                                free(*out);
+                        *out = strdup(v);
+                        if (!*out)
+                                return log_oom();
```

Comment:
OK, so the function should find the last PRETTY_NAME, and return the strdup()ed value in the output argument. The code will be nicer if you use a temporary variable to store the value, and simply strdup() it at the end. E.g.:
```
    return free_and_strdup(out, val);
```